servicestack-client/index.d.ts

```diff
@@ -0,0 +1,155 @@
+// Type definitions for servicestack-client v0.0.17
+// Project: https://servicestack.net/
+// Definitions by: Demis Bellot <https://github.com/mythz/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference path="../isomorphic-fetch/index.d.ts" />
+
+declare namespace ServiceStack {
+export interface IReturnVoid {
+    createResponse(): any;
+}
+export interface IReturn<T> {
+    createResponse(): T;
+}
+export class ResponseStatus {
+    errorCode: string;
+    message: string;
+    stackTrace: string;
+    errors: ResponseError[];
+    meta: {
+        [index: string]: string;
+    };
+}
+export class ResponseError {
+    errorCode: string;
+    fieldName: string;
+    message: string;
+    meta: {
+        [index: string]: string;
+    };
+}
+export class ErrorResponse {
+    responseStatus: ResponseStatus;
+}
+export interface ISseCommand {
+    userId: string;
+    displayName: string;
+    channels: string;
+    profileUrl: string;
+}
+export interface ISseHeartbeat extends ISseCommand {
+}
+export interface ISseJoin extends ISseCommand {
+}
+export interface ISseLeave extends ISseCommand {
+}
+export interface ISseUpdate extends ISseCommand {
+}
+export interface ISseConnect extends ISseCommand {
+    id: string;
+    unRegisterUrl: string;
+    heartbeatUrl: string;
+    updateSubscriberUrl: string;
+    heartbeatIntervalMs: number;
+    idleTimeoutMs: number;
+}
+export interface IReconnectServerEventsOptions {
+    url?: string;
+    onerror?: (...args: any[]) => void;
+    onmessage?: (...args: any[]) => void;
+    errorArgs?: any[];
+}
+/**
+ * EventSource
+ */
+export enum ReadyState {
+    CONNECTING = 0,
+    OPEN = 1,
+    CLOSED = 2,
+}
+export interface IEventSourceStatic extends EventTarget {
+    new (url: string, eventSourceInitDict?: IEventSourceInit): IEventSourceStatic;
+    url: string;
+    withCredentials: boolean;
+    CONNECTING: ReadyState;
+    OPEN: ReadyState;
+    CLOSED: ReadyState;
+    readyState: ReadyState;
+    onopen: Function;
+    onmessage: (event: IOnMessageEvent) => void;
+    onerror: Function;
+    close: () => void;
+}
+export interface IEventSourceInit {
+    withCredentials?: boolean;
+}
+export interface IOnMessageEvent {
+    data: string;
+}
+export class ServerEventsClient {
+    channels: string[];
+    options: any;
+    eventSource: IEventSourceStatic;
+    eventSourceUrl: string;
+    updateSubscriberUrl: string;
+    eventSourceStop: boolean;
+    constructor(baseUrl: string, channels: string[], options?: any, eventSource?: IEventSourceStatic);
+    onMessage(e: IOnMessageEvent): void;
+    reconnectServerEvents(opt?: any): IEventSourceStatic;
+    invokeReceiver(r: any, cmd: string, el: Element, msg: string, e: any, name: string): void;
+    updateChannels(channels: string[]): void;
+}
+export class HttpMethods {
```

Comment:
Because packaging types with npm package [doesn't work with JSPM yet](https://github.com/Microsoft/TypeScript/issues/6012) from what I understand publishing `@types` separately was the only solution. I've just learnt of another approach which I still have to try out, but otherwise I'd need to publish the type defs to `@types`.